ipa-core/src/report/matchkey.rs

```diff
@@ -0,0 +1,283 @@
+            debug_assert_eq!(out.len(), usize::from(self.encrypted_len()));
+            Ok(out)
+        }
+
+        /// # Errors
+        /// If there is a problem encrypting the report.
+        pub fn encrypt_to<R: CryptoRng + RngCore, B: BufMut>(
+            &self,
+            key_id: KeyIdentifier,
+            key_registry: &impl PublicKeyRegistry,
+            rng: &mut R,
+            out: &mut B,
+        ) -> Result<(), InvalidReportError> {
+            let info = Info::new(
+                key_id,
+                self.epoch,
+                self.event_type,
+                HELPER_ORIGIN,
+                self.site_domain.as_ref(),
+            )?;
+
+            let mut plaintext_mk = GenericArray::default();
+            self.match_key.serialize(&mut plaintext_mk);
+
+            let (encap_key_mk, ciphertext_mk, tag_mk) =
+                seal_in_place(key_registry, plaintext_mk.as_mut(), &info, rng)?;
+
+            out.put_slice(&encap_key_mk.to_bytes());
+            out.put_slice(ciphertext_mk);
+            out.put_slice(&tag_mk.to_bytes());
+            out.put_slice(&[u8::from(&self.event_type)]);
+            out.put_slice(&[key_id]);
+            out.put_slice(&self.epoch.to_le_bytes());
+            out.put_slice(self.site_domain.as_bytes());
+
+            Ok(())
+        }
+    }
+    #[test]
+    fn test_swift_encryption() {
+        let pk = hex::decode("92a6fb666c37c008defd74abf3204ebea685742eab8347b08e2f7c759893947a")
+            .unwrap();
+        let sk = hex::decode("53d58e022981f2edbf55fec1b45dbabd08a3442cb7b7c598839de5d7a5888bff")
+            .unwrap();
+        let key_registry = KeyRegistry::from_keys([KeyPair::from((
+            IpaPrivateKey::from_bytes(&sk).unwrap(),
+            IpaPublicKey::from_bytes(&pk).unwrap(),
+        ))]);
+        let relative_path = "../test_data/fbinfra_integration/ios_encrypted_matchkey.txt"; // replace with your relative path
+                                                                                           // let absolute_path = join(relative_path);
+        let mut file = File::open(relative_path).expect("File not found");
+        let mut bytes = Vec::new();
+        file.read_to_end(&mut bytes).expect("Error reading file");
+        let report = EncryptedMatchKeyReport::<_>::from_bytes(bytes.as_slice()).unwrap();
+
+        let expected = MatchKeyReport {
+            match_key: Replicated::new(
+                BA64::try_from(1_u128).unwrap(),
+                BA64::try_from(2_u128).unwrap(),
+            ),
+            event_type: EventType::Source,
+            epoch: 0,
+            site_domain: String::from("www.meta.com"),
+        };
+        let result = report.decrypt(&key_registry).unwrap();
+        assert_eq!(expected, result);
+    }
+
+    #[test]
+    fn test_matchkey_decrypt_encrypt() {
+        let mut rng = thread_rng();
+        let pk = hex::decode("92a6fb666c37c008defd74abf3204ebea685742eab8347b08e2f7c759893947a")
```

Comment:
maybe you could avoid copy-pasting the same key material and set it up once as a constant for all tests.

but probably if you could use `GenericReport` you don't need to test the encryption roundtrip